core/schema/container.go

```diff
@@ -184,6 +184,13 @@ func (s *containerSchema) withExec(ctx *router.Context, parent *core.Container,
 	return parent.WithExec(ctx, s.gw, s.baseSchema.platform, args.ContainerExecOpts)
 }
 
+func (s *containerSchema) withDefaultExec(ctx *router.Context, parent *core.Container) (*core.Container, error) {
```

Comment:
I like this - handling it at the API layer keeps the `core.Container` type from having to just-in-time modify itself.